README.adoc

```diff
@@ -1,160 +1,233 @@
+[source,shell]
+----
+curl -G https://start.vertx.io/starter.zip -d "groupId=io.vertx.howtos" -d "artifactId=protobuf-eventbus-howto" -d "packageName=io.vertx.howtos.protobuf.eventbus" -d "vertxDependencies=vertx-hazelcast" -d "jdkVersion=11" -d "buildTool=maven" --output protobuf-eventbus-howto.zip
+unzip -d protobuf-eventbus-howto protobuf-eventbus-howto.zip
+----
+====
 
-=== What you need
+Before coding, we need to make some adjustments to the build file:
 
-- List all required tools to edit, build and run the how-to.
-- Do not hesitate to point to third-party resources (e.g., if you need Docker, say why you need it and point to its installation documentation).
+* configuring a custom Vert.x `Launcher` class (that will be used as entry point when running the executable JAR)
+* adding a dependency to Protocol Buffers
+* configuring Maven plugins to generate message classes from a `.proto` file.
 
-=== Create a project
+Here is the content of the `pom.xml` file you should be using:
 
-- Explain how to create the project.
-- Show the Maven / Gradle build descriptors.
-- For any third-party tool (e.g., running a database with Docker), also have a subsection to explain this.
+[source,xml,role="collapsed"]
+.Maven `pom.xml`
+----
+include::pom.xml[]
+----
 
-=== Implementation sections
+== Implementation of the application
 
-- Make one or multiple sections to show the implementation.
-- Make use of Asciidoc source code comments to highlight specific parts of the code.
-- The code must be included using https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#source-code[Asciidoc directives]. Remember that you can include fragments of code by using delimiter tags.
+=== Definition of the messages
 
-=== Running the application
+In `src/main/proto/greetings.proto`, we define:
 
-- Always show the various options to run the application.
+* a `GreetingRequest` which holds a `name`, and
+* a `GreetingReply` which holds a `message`
 
-=== Summary
+[source,protobuf]
+.greetings.proto
+----
+include::src/main/proto/greetings.proto[]
+----
 
-- Briefly recapitulate what was covered and learned in the how-to.
+=== Receiver verticle
 
-=== See also
+The receiver verticle registers a consumer on the Event Bus.
+When a request is received:
 
-- If possible, link to other how-tos.
-- If judicious, link to Vert.x or third-party documentation.
+. the request is printed to the console along with its system _hashcode_
+. a reply is generated
+. the reply is printed to the console along with its system _hashcode_
+. the reply is sent
 
-== How is the website being rendered?
+[source,java]
+.ReceiverVerticle.java
+----
+include::src/main/java/io/vertx/howtos/protobuf/eventbus/ReceiverVerticle.java[]
+----
 
-The website is being rendered using Jekyll and the Asciidoctor extension for Jekyll.
+NOTE: The system _hashcode_ is printed so that when we run the application in a single virtual machine, we can see whether objects are duplicated or not by the Event Bus.
```

Comment:
```suggestion
NOTE: The system _hash code_ is printed so that when we run the application in a single virtual machine, we can see whether objects are duplicated or not by the Event Bus.
```